src/Xamarin.Android.Build.Tasks/Tasks/GeneratePackageManagerJava.cs

```diff
@@ -323,7 +361,9 @@ void AddEnvironment ()
 					JniAddNativeMethodRegistrationAttributePresent = appConfState != null ? appConfState.JniAddNativeMethodRegistrationAttributePresent : false,
 					HaveRuntimeConfigBlob = haveRuntimeConfigBlob,
 					NumberOfAssembliesInApk = assemblyCount,
-					BundledAssemblyNameWidth = assemblyNameWidth + 1,
+					BundledAssemblyNameWidth = assemblyNameWidth,
+					NumberOfAssemblyBlobsInApk = 2, // Until feature APKs are a thing, we're going to have just two blobs in each app
```

Comment:
…which implies two blobs.  What are the blobs?  Or is this related to the previous "blob ID" comment? https://github.com/xamarin/xamarin-android/pull/6311/files#diff-9b3577c08294fc97880076f163cbbde1e3fb46a0d20dde1a89cbf94ab932a1f7R347

> The very first blob (ID 0) **must** contain an index of all the assemblies included in the application

At this point in my review, I haven't seen anything about the *structure* of the blobs, and how it relates to the structure of `.aab` and `.apk` files…

Reply:
Related: we need an "overview" of how this works and how all the pieces fit together, either in the commit message or in a `README.md`.

Reply:
The structure is described [here](https://github.com/xamarin/xamarin-android/pull/6311/files#diff-920537864b4d4fbcacfad5ebfdcfd73470855176b83f4c4592d604a7057b7d3dR111-R131) and I'm going to add a doc once everything is certain to be correct (which it should be now, hopefully). I'm working on adding blob support to some utilities